autoload/jq.vim

```diff
@@ -0,0 +1,26 @@
+""
+" Refer https://github.com/vim/vim/blob/75e27d78f5370e7d2e0898326d9b080937e7b090/runtime/scripts.vim#L33-L71
+function! jq#shebang() abort
+    let s:line1 = getline(1)
+
```

Comment:
This all seems a little overly complicated. What exactly are we trying to achieve can you give me some samples of what shebangs aren't working for you?

Reply:
> `#!/usr/bin/env -S sh -c 'echo jq $0'` will still be recognized to jq.
oh so it's being too greedy

> Now this code is referenced from vim's code which only recognize `#!/the/path/jq`, `#!/the/path/env jq #!/the/path/env -S jq XXX`

All of the code you referenced is no longer in the master branch. It was moved to a function.

Reply:
> oh so it's being too greedy

In my opinion, it makes code more robust.

> All of the code you referenced is no longer in the master branch

Yes, It doesn't affect if it can work.